test/Microsoft.AspNetCore.Mvc.Razor.Test/RazorPageTest.cs

```diff
@@ -472,6 +472,23 @@ public void IsSectionDefined_ThrowsIfPreviousSectionWritersIsNotRegistered()
         }
 
         [Fact]
+        public async Task EnsureRenderedBodyOrSections_SucceedsIfRenderBodyIsNotCalledFromPage_AndNoSectionsAreDefined_AndBodyIgnored()
+        {
+            // Arrange
+            var path = "page-path";
+            var page = CreatePage(v =>
+            {
+            });
+            page.Path = path;
+            page.BodyContent = new HtmlString("some content");
+            page.IgnoreBody();
+
+            // Act & Assert (does not throw)
+            await page.ExecuteAsync();
+            page.EnsureRenderedBodyOrSections();
```

Comment:
I'm not sure I understand what you mean here, this test wont write anything to the page
because body is ignored and no sections defined